packages/esm-form-entry-app/src/app/form-data-source/form-data-source.service.ts

```diff
@@ -114,6 +117,56 @@ export class FormDataSourceService {
     );
   }
 
+  public getMostRecentObsValueBefore(date: string, patientUuid: string) {
```

Comment:
See earlier comments on the parameters here (and check the model signature from the ticket, since this doesn't implement the requirements. Also, why would we pass the date as a string here? Just take a JS date object...